### ISSUE TITLE
Compile code with Java 6 compatibility

### ISSUE THREAD

**Issue by AdrianBuza** [odaata/HisTEI#13]
Oxygen still supports Java 6, especially on OS X. The extension should be compiled with Java 6 compatibility flags.
javac -source 1.6 -target 1.6

Otherwise you'll get:
java.lang.UnsupportedClassVersionError: info/histei/HTExtensionsBundle : Unsupported major.minor version 51.0 

Regards,
Adrian

**Comment by odaata** [comment]
Although Oxygen still supports Java 6, I decided to use Java 7 for the HisTEI framework and only support that version or higher. I did this for multiple reasons: Java 6 is not receiving security updates/patches anymore, version 7 has some nice new language features and I was interested in using NIO to track files (though in the end I didn't actually use it). Can you upgrade your Java to version 7? Also, if you're on a Mac, do note that you need to download the Oxygen package that includes the latest version of Java (down on the right-hand side of the screen). This is also explained on the Installation page of the Wiki.

**Comment by AdrianBuza** [comment]
I'm part of the Oxygen support team (I should have mentioned this before) and we sometimes receive problem reports from Oxygen users with UnsupportedClassVersionError for your addon. These are users running Oxygen on OS X where we still recommend using Java 6. It's true that there's also a distribution of Oxygen with Java 8 (we've upgraded it), but right now we only recommend it in special cases.

Anyway, what I was proposing was to configure the Java compiler (javac) compatibility flags to keep the generated Java bytecode of the compiler compatible with Java 6.
So, you can still use the Java 7 or 8 JDK for compiling, but with the Java 6 flags (-source 1.6 -target 1.6).

**Comment by odaata** [comment]
Yes, I could do that, the only problem is that I've been using new language features available in Java 7, so my source is really 1.7 and the compiler won't compile it down to 1.6. I'll re-open this issue as something to work on though. In the meantime, I've been telling Mac users to download the Oxygen version with the latest SDK.

**Comment by sherlok235** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
Farman2018, @MajestyIndestructible, @boranbekdas, @LakkyTub, @kguler10, @xSkyScraper, @patator62180, @Kalftrue, @gotomoro, @YCHkevin, @NicholasGobettiOshiro, @PREDOKX2020, @xmaskot1, @sq78, @jamievanschuerbeek, @dkloving, @yasin-rsln, @xdf-git, @salih64, @Lokorus

**Comment by sherlok235** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
eliot, @OlgaKhinevich, @oceceli, @StormFire2k, @MrsProxima, @kaanmutlus, @greenday0989, @SamuelSarle, @ilesoviy, @hibyemy, @laurentiu1963, @vaneche, @kabilApp, @HarrisonMclean, @erenakkaya, @brkfdkr, @kkidd01, @vicatz, @123456982, @CanRau
